crates/turborepo-cache/src/lib.rs

```diff
@@ -105,12 +109,43 @@ pub struct CacheHitMetadata {
     pub time_saved: u64,
 }
 
-#[derive(Clone, Debug, Default)]
+#[derive(Serialize, Deserialize, Deserializable, Debug, PartialEq, Eq, Clone, Copy)]
+pub struct CacheActions {
+    pub read: bool,
+    pub write: bool,
+}
+
+impl CacheActions {
+    pub fn should_use(&self) -> bool {
+        self.read || self.write
+    }
+}
+
+#[derive(Serialize, Deserialize, Deserializable, Debug, PartialEq, Eq, Clone, Copy, Default)]
+pub struct CacheConfig {
+    pub local: CacheActions,
+    pub remote: CacheActions,
+}
```

Comment:
Minor code structure issue: I think the ser/de implementations should live where we care/use them in the `turborepo_lib::config` module. Also, do we want to support users setting this value in `.turbo/config.json`, if not then we should just skip deserializing this field and avoid the unused implementation.